flang/lib/Optimizer/CodeGen/CodeGen.cpp

```diff
@@ -67,8 +68,41 @@ static constexpr unsigned defaultAlign = 8;
 static constexpr unsigned kAttrPointer = CFI_attribute_pointer;
 static constexpr unsigned kAttrAllocatable = CFI_attribute_allocatable;
 
-static inline mlir::Type getLlvmPtrType(mlir::MLIRContext *context) {
-  return mlir::LLVM::LLVMPointerType::get(context);
+static inline unsigned getAllocaAddressSpace(mlir::ModuleOp module) {
+  if (mlir::Attribute addrSpace =
+          mlir::DataLayout(module).getAllocaMemorySpace())
+    return addrSpace.cast<mlir::IntegerAttr>().getUInt();
```

Comment:
`cast` utils in `mlir::Attribute` are [deprecated](https://github.com/llvm/llvm-project/blob/c69ec700adec315b3daa55742f2ef655242fa297/mlir/include/mlir/IR/Attributes.h#L50C1-L50C1).